examples/tutorials/pruning_bert_glue.py

```diff
@@ -0,0 +1,534 @@
+"""
+Pruning Transformer with NNI
+============================
+
+Workable Pruning Process
+------------------------
+
+Here we show an effective transformer pruning process that NNI team has tried, and users can use NNI to discover better processes.
+
+The entire pruning process can be divided into the following steps:
+
+1. Finetune the pre-trained model on the downstream task. From our experience,
+   the final performance of pruning on the finetuned model is better than pruning directly on the pre-trained model.
+   At the same time, the finetuned model obtained in this step will also be used as the teacher model for the following
+   distillation training.
+2. Pruning the attention layer at first. Here we apply block-sparse on attention layer weight,
+   and directly prune the head (condense the weight) if the head was fully masked.
+   If the head was partially masked, we will not prune it and recover its weight.
+3. Retrain the head-pruned model with distillation. Recover the model precision before pruning FFN layer.
+4. Pruning the FFN layer. Here we apply the output channels pruning on the 1st FFN layer,
+   and the 2nd FFN layer input channels will be pruned due to the pruning of 1st layer output channels.
+5. Retrain the final pruned model with distillation.
+
+During the process of pruning transformer, we gained some of the following experiences:
+
+* We using :ref:`movement-pruner` in step 2 and :ref:`taylor-fo-weight-pruner` in step 4. :ref:`movement-pruner` has good performance on attention layers,
+  and :ref:`taylor-fo-weight-pruner` method has good performance on FFN layers. These two pruners are all some kinds of gradient-based pruning algorithms,
+  we also try weight-based pruning algorithms like :ref:`l1-norm-pruner`, but it doesn't seem to work well in this scenario.
+* Distillation is a good way to recover model precision. In terms of results, usually 1~2% improvement in accuracy can be achieved when we prune bert on mnli task.
+* It is necessary to gradually increase the sparsity rather than reaching a very high sparsity all at once.
+
+Experiment
+----------
+
+Preparation
+^^^^^^^^^^^
+Please set ``dev_mode`` to ``False`` to run this tutorial. Here ``dev_mode`` is ``True`` by default is for generating documents.
+
+The complete pruning process takes about 8 hours on one A100.
+"""
+
+dev_mode = True
+
+# %%
+# Some basic setting.
+
+if not dev_mode:
```

Comment:
When dev mode is true, can we run the code but only for one mini-batch?

Reply:
fixed, run a few mini-batches for sanity check under dev-mode